Epic 3 - ReportApp/ReportApp/Services/Activity/ActivityReportGeneratorService.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            var template = _templateLoader.LoadTemplate("./Templates/ActivityReportTemplate.xlsx");
+            var configuration = _configurationLoader.LoadConfiguration("./ReportConfigurations/Activity.json");
+
+            var model = _reportSerializer.DeserializeReportModel(jsonPath);
+            FillSettings(model);
+            FillReportDataFromModel(template, configuration, model);
+            FillHeader(template, configuration);
+
+            template.Generate();
+            template.SaveAs(outputPath);
+        }
+
+        private void FillSettings(ActivityReportModel model)
+        {
+            Admin? generatedByAdmin = null;
+            Client? generatedByClient = null;
+            var generatedFor = $"{model.ReportGeneratedFor.FirstName} {model.ReportGeneratedFor.LastName}";
+
+            if (model.GeneratedByAdmin != null)
+            {
+                generatedByAdmin = model.GeneratedByAdmin;
+            }
+            else
+            {
+                generatedByClient = model.GeneratedByClient;
+            }
+
+            Settings = new ActivityReportSettings
+            {
+                GeneratedByAdmin = generatedByAdmin,
+                GeneratedByClient = generatedByClient,
+                GeneratedFor = generatedFor,
+            };
+        }
+
+        private void FillHeader(XLTemplate template, ActivityReportConfiguration configuration)
+        {
+            if (Settings != null)
+            {
+                var isGeneratedByAdmin = Settings.GeneratedByAdmin != null ? true : false;
+                template.AddVariable("GeneratedFor", Settings.GeneratedFor);
+
+                if (isGeneratedByAdmin)
+                {
+                    template.AddVariable("GeneratedBy",
+                        $"{Settings.GeneratedByAdmin.FirstName} {Settings.GeneratedByAdmin.LastName} (Admin)");
+                }
+                else
+                {
+                    template.AddVariable("GeneratedBy",
+                        $"{Settings.GeneratedByClient.FirstName} {Settings.GeneratedByClient.LastName} (Client)");
+                }
+            }
+        }
+
+        private void FillReportDataFromModel(XLTemplate template, ActivityReportConfiguration configuration, ActivityReportModel model)
+        {
+            var worksheet = template.Workbook.Worksheets.First();
+            worksheet.SetShowGridLines(false);
+
+            var firstDataRow = configuration.DefaultRow;
+            var firstDataColumn = configuration.FirstColumn;
+
+            var groupAmount = 2;
+            var lastDataColumn = configuration.LastColumn;
+
+            if (model.GeneratedByAdmin == null)
+            {
+                groupAmount = 1;
+                lastDataColumn -= 3;
+                worksheet.Range(configuration.FirstRowForDynamicGroup, lastDataColumn + 1, configuration.FirstRowForDynamicGroup, configuration.LastColumn)
```

Comment:
Looks like this method FillsData and making something else. Single responsibility